### ISSUE TITLE
Calling apiExecute('system', ‘echo hello’) while being in an conference makes audio clip

### ISSUE THREAD

**Issue by daviderenger** [signalwire/freeswitch#1210]
We just made it through a rough period of troubleshooting mysterious audio problems and I wanted to share our findings since I think it is a bug or at least a very unexpected behavior with potentially catastrophic results.

The problem we had was that we had glitches in the audio when being in a conference with Verto. For the client it showed as RTP-packets with a delta around 100ms and a marker bit arriving from time to time.

We probably always had this problem but when the frequency was low it was not that noticeable. As our load increased we saw increased packets like this and the sound started to be pretty bad. We thought it had to do with load of the servers of course and changed to more powerful servers but the problem stayed exactly the same even with a very low load. 

Reading email threads and forum posts about problems like this always points towards timer problems and that using physical hardware is advised. We tried everything suggested, using different timers, changing udp-buffer sizes etc etc but no difference.

When troubleshooting this we had problems recreating the issue on staging environments even though we bombarded that server with many more calls that we had on the production servers. We also tried to load down the network and a lot of other things.

And finally we found the root cause of the problem! In a very specific call scenario we were using apiExecute('system',... for a trivial task, probably some copy/paste from an example. And whenever apiExecute('system',... is executed on any thread all call legs in all conferences in the entire FreeSwitch gets an audio glitch shown as a RTP-packet with ~100ms delta and a marker bit.

So the problem didn't have anything to do with load or number of calls, it happened every time that function is called and the reason we had more problems as load increases was that it happened more frequently.

We could reproduce the problem simply by calling ‘system echo "hello"’ from fs_cli.

The solution for us was very simple: stop using apiExecute('system',...

Maybe this is a bug somehow so it can be fixed or maybe it is expected but I feel that if this is expected behavior a warning should be added to the documentation. 

As a side note we had unexpected high system load during this time period that we really couldn't explain. As you can see that load dropped to almost nothing when the patch was applied at 14:53.

![loadChart](https://user-images.githubusercontent.com/18427393/119620532-62678f00-be05-11eb-9890-c87da4b42fbf.png)

**To Reproduce**
1. Dial in to a conference with Verto
2. Setup wireshark so you can capture the RPT packets
3. Call “system echo "hello" “ from fs_cli
4. See that a RPT-packet with a high delta and a marker bit arrives in wireshark (and you can also here a glitch sound if you have a counterpart with constant audio)

**Expected behavior**
No marker bits nor audio glitches in ongoing conferences when calling system.

**Package version or git hash**
 - 1.10.6 with https://github.com/signalwire/freeswitch/commit/7fc5d9a7c36337b15a5c7044bdaecdc82d95d94b
cherry-picked

**Comment by dragos-oancea** [comment]
do you have threaded-system-exec set to true in autoload_configs/switch.conf.xml ?

**Comment by daviderenger** [comment]
No we don't have that, but it sounds like a good thing for this. We will try it out. Cant really find any documentation of it, does it have any drawbacks and why is it not the default mode?

**Comment by snaptel** [comment]
@daviderenger did you  test it with the threaded-system-exec option set to true?

**Comment by ar45** [comment]
`threaded-system-exec` is only effective for application `system` which is defined in mod_dptools. for api system defined in mod_commands it always uses `popen()` from stdlib in the same thread as the calling code.

@daviderenger  When executing an API command from fs_cli is different than running elsewhere, as it runs in it's own thread within mod_event_socket.

**Comment by ar45** [comment]
@daviderenger Can you test while executing your system command using a lower priority? When you run your command within FS as an api it forks a sub process with the same priority as FS itself is running. 

Try running with `system chrt —other 0 yourCommand`

**Comment by daviderenger** [comment]
We will test all written above, also test with bg_system. The urgent problem is already solved since we didn't really need the system call.

We also noticed that we need to have some load (many threads?) for this to happen. But again, we will try the flags and the different calls and report back!

**Comment by daviderenger** [comment]
Sorry for the late response but now we have tested all of the above suggestions.

We added threaded-system-exec to the conf and then executing with apiExcecute from Javascript the following:
system
bg_system
system chrt —other 0

And the result is the same for all; on every call we get a jump in the stream and a marker bit (and an audio glitch) in all ongoing conferences.

**Comment by ar45** [comment]
@daviderenger can you show your CPU usage for during that period? I was only able to reproduce that when reaching hi 100% of cpu.

**Comment by daviderenger** [comment]
@ar45 you are right, I forgot to mention that you need some load, but not 100% CPU. It seems to be more related to how many threads you have. For this latest test we had about 30-40% CPU load and 60 ongoing calls.

When we investigated the original problem we doubled the instance type on change with no luck (lowering CPU from 40% to 20%). So I would say number of threads is the thing that effects this behavior.

**Comment by dragos-oancea** [comment]
how about if you set chan var spawn_instead_of_system to "true". master branch.

**Comment by snaptel** [comment]
@dragos-oancea 
just curious, why would spawn_instead_of_system be expected to be less of a problem, wouldn't it be more load on the system? the reason I ask is because I experienced similar issues in the past but can't reproduce it so no way of testing this new option.
